documentation/Software Requirements Specification.md

```diff
@@ -0,0 +1,234 @@
+<b>Main scenario:</b>
+* User selects item “Employee”;
+* Application displays list of Employee.
+
+![](employee_listt.jpg)
+Pic. 3.1 View the Employee list.
+
+<b>The list displays the following columns:</b>
+* ID – unique employee number;
+* Name - first name and last name
+* Date of birth - employee's date of birth
+* Department - the department in which the employee consists
+* Positions - employee positions
+* Specialization - skills that the employee has
+* Experience - work experience
+* Wages - employee's salary
+
+<b>Filtering by date:</b>
+* In the employee list view mode, the administrator sets a filter by date and clicks the update list button (to the right of the date input field).;
+* The application will show employees only during a certain period of the date of birth.
+
+<b>Restrictions:</b>
+* Start date of the period should be less then end date of the period;
+* If start date is blank, then filtering by end date only.
+* If end date is blank, then filtering by start date only.
+* Updating data after selecting the filtering conditions is carried out by pressing the “Refresh” button.
+
+### Add employee
+
+<b>Main scenario:</b>
+* Administrator clicks the “Add” button in the employee list view mode;
+* Application displays form to enter employee data;
+* Administrator enters employee data and presses “Save” button;
+* If any data is entered incorrectly, incorrect data messages are displayed;
+* If entered data is valid, then record is adding to database;
+* If error occurs, then error message is displaying;
+* If new employee record is successfully added, then list of employees with added records is displaying.
+
+<b>Cancel operation scenario:</b>
+* Administrator clicks the “Add” button in the employees list view mode;
+* Application displays form to enter employees data;
+* User enters employees data and presses “Cancel” button;
+* Data don’t save in data base, then list of employees records is displaying to administrator.
+* If the administrator selects the menu item "Department ” or "Employee", the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+![](add_employee.jpg)
+<br>Pic. 3.2 Add employee.
+
+<b>When adding a employee, the following details are entered:</b>
+* Name - first name and last name
+* Date of birth - employee's date of birth
+* Department - the department in which the employee consists
+* Positions - employee positions
+* Specialization - skills that the employee has
+* Experience - work experience
+* Wages - employee's salary
+### Edit employee
+
+<b>Main scenario:</b>
+* Administrator clicks the “Edit” button in the employee list view mode;
+* Application displays form to enter employee data;
+* Administrator enters employees data and presses “Save” button;
+* If any data is entered incorrectly, incorrect data messages are displayed;
+* If entered data is valid, then edited data is added to database;
+* If error occurs, then error message is displaying;
```

Comment:
It would be great to see a mockup with error messages.

Reply:
Fixed